tempora/timing.py

```diff
@@ -220,45 +232,54 @@ class BackoffDelay(collections.abc.Iterator):
     True
     """
 
-    delay = 0
-
     factor = 1
     "Multiplier applied to delay"
 
-    jitter = 0
-    "Number or callable returning extra seconds to add to delay"
+    jitter: collections.abc.Callable[[], float]
+    "Callable returning extra seconds to add to delay"
 
     @jaraco.functools.save_method_args
-    def __init__(self, delay=0, factor=1, limit=float('inf'), jitter=0):
+    def __init__(
+        self,
+        delay: float = 0,
+        factor=1,
+        limit: collections.abc.Callable[[float], float] | float = float('inf'),
+        jitter: collections.abc.Callable[[], float] | float = 0,
+    ) -> None:
         self.delay = delay
         self.factor = factor
-        if isinstance(limit, numbers.Number):
+        if isinstance(limit, (numbers.Number, int, float)):
             limit_ = limit
 
-            def limit(n):
+            def limit_func(n: float, /) -> float:
                 return max(0, min(limit_, n))
 
-        self.limit = limit
-        if isinstance(jitter, numbers.Number):
+        else:
+            limit_func = limit
+        self.limit = limit_func
+        if isinstance(jitter, (numbers.Number, int, float)):
             jitter_ = jitter
 
-            def jitter():
+            def jitter_func() -> float:
                 return jitter_
 
-        self.jitter = jitter
+        else:
+            jitter_func = jitter
+
+        self.jitter = jitter_func
```

Comment:
I don't love how these code blocks are getting bigger. I'll probably refactor into a singledispatch wrapper. But this is fine for now.